### ISSUE TITLE
Dashboard

### ISSUE THREAD

**Issue by netluca1** [tananaev/traccar#2860]
Create a dashboard area that is displayed by pressing a button in the top bar
Is possible create with metabase
http://www.metabase.com/
https://github.com/metabase/metabase

![cruscotto](https://cloud.githubusercontent.com/assets/23439828/22491576/c3a4db4e-e824-11e6-877a-e95b87155730.png)

**Comment by hackerunet** [comment]
I can suggest you to build your own interface using whatever you may want to use, any language, then use traccar server for decodification and GPS communication integration into your new system, and integrate the database and all queries with your database schema, is not too hard, in fact is very easy and step forward process.

**Comment by djbo1982** [comment]
Eso se ve muy bien @hackerunet, are you speak spanish?

**Comment by hackerunet** [comment]
si, hablo español.

**Comment by tananaev** [comment]
I don't see much value in the dashboard. Please provide more details on what you want to put there and what's the business case for it.

**Comment by cdmarioe** [comment]
Anybody has implented metabase/traccar?

**Comment by MaikZaavedra** [comment]
@hackerunet podrias asesorarme con metabase yo tambien quiero implementarlo a traccar

**Comment by hackerunet** [comment]
What exactly do you need?

**Comment by hackerunet** [comment]
this seems to be forgotten.

**Comment by oliv3** [comment]
@hackerunet you could try grafana to create nice dashboards

**Comment by tananaev** [comment]
This is not your ticket. Why are you deciding for someone else that it can be closed?

**Comment by hackerunet** [comment]
Sorry i didn't meant to. is just that the ticket is super old. also im not
an admin or the owner im just trying to help to get some tickets closed,
tickets that could be closed because the author did nothing for very long
time, but i will stay out of the community then. im sorry.

**Comment by tananaev** [comment]
I think we'll keep this one for now. It's a fairly popular request. Event if there is no plan to implement it at the moment, it doesn't mean it's never going to be implemented.

**Comment by djbo1982** [comment]
Well... i can paid is some body want to develop some similar of this dashboard

**Comment by netluca1** [comment]
Hi, I'm always interested in this implementation, maybe as you mentioned @ djbo1982 we could reach a contribution to integrate

**Comment by oliv3** [comment]
I also think this would be a great addition to the product.
In the meantime, I'm working on a dashboard like this, to be integrated in Grafana.
So basically, which metrics/reports would you like to see ?
Has anyone specifications ?

**Comment by djbo1982** [comment]
@oliv3 @cdmarioe  @netluca1  well, i work with Car managed system (fleet management software for vehicles), we need to analyzed the metric/reports to know what we can integrate, but sound great can have the option to personalize what show the Dashboard.

Maybe we can the news API of traccar a new programming language for this

I know, the most basic show are, the list of car with overspeed, out and in geofence, maybe maintenance car list expired, list of off line cars... but not depend only of me, if another persons want to paid to for this develop.

**Comment by netluca1** [comment]
Thanks @ oliv3 has x your interest in the change.
I think that the example in the photo proposed is very exhaustive.
In fact, they put everything that could be of interest to all corporate fleets.
I think it is possible to obtain all the necessary data using both the devices that send the device and those calculated
In fact it is proposed:
Fleet Status (current General overview of the moment): Moving Vehicle / Vehicle stationary with Engine On / Vehicle Stationary
Daily Activity: Similar to the previous event but with reference to the Day.
I personally would also add the possibility to choose whether Daily / Weekly / Monthly
Covered distances (KM) always with reference: Daily / Weekly / Monthly
Time of use Hours of driving always with reference: Daily / Weekly / Monthly
Average fuel consumption
Fuel wasted very important data coming from the engine on and half stopped.
Infringements Section: fines derived from excesses of speed or from the tachograph that marks everything (guided driving hours, Excess driving hours, speed, etc.)
To complete I would also prepare other 2 that is Temperature and Humidity.
Also to make it all back I would give the possibility to the administrator or who for him to decide what to see if one or more of all these functions.
This is my thought I hope it could have been helpful cmq every other suggestion is always welcome

Greetings Luca

**Comment by hackerunet** [comment]
I think the best for this could be to develop an Angular 5+ client to consume the traccar API, implement JWT token for authentication and with that the dashboard could have all those features.

**Comment by djbo1982** [comment]
@hackerunet Yes, because that i ask for another programming language, because we need to use the Traccar API, Tokens and others.

@netluca1 is very important have option to choice what we go to have in the dashboard, because we need to think in different projects not only in own projects...

Lest try to make a simple list of what we want and need, with priorities and lest wait how want to develop this @oliv3 or @hackerunet, and we can use this Github to inform our step, but for develop we need a space to communicate with the programming persons, like Trello to assign task and review the advances.

Is very important put all here for the Traccar project can use in some moment our advances to integrate in the principal project

**Comment by netluca1** [comment]
Hi @hackerunet  your idea is to develop something outside the program?
Honestly, the thing could be nice and independent but my experience with similar things testifies to me that it is not user friendly.
The end user wants everything he does not want to turn from one side to the other with other programs.
The experience is to have everything at hand ready to be consulted at will

**Comment by djbo1982** [comment]
@netluca1 my idea is same of you!

Use this with Traccar because we want an only one interface, i love the simple of the traccar manager views... is the best, but my users wants a more friendly interface, but is tru if we go to develop other characteristics of traccar, we need to think in all users and this need to be approved for the owners of the project! 

But, if we can’t have this approved we need to think in do a way to integrate in traccar and persist in the futures releases 

As I say, I work with software and we need to think in all possibilities of this implementation

**Comment by hackerunet** [comment]
ok, @netluca1 @djbo1982  it doesn't have to be outside of traccar, today's development is projected to have mobility so software developers are quitting developing complex backends with no restfull api, almost everything today has a restfull API, traccar is also the case, we could think in create the client using angular and a material based dashboard and it will be user friendly, believe me, also it would be mobile friendly, I'm totally sure that angular 5+ makes user experience great, but also is not up to the language itself but to the implementation.
For example I built a UI using Sencha like 6 years ago, and it was great for my customers because I implemented my own logic to it, Traccar was also built using sencha but less friendly so my customers loved the UI I built, so this is just to show you the point, which the user experience relies on the implementation not in the language.
Also why angular and not Vue js for example? because every JS framework relies on the Components library and I think angular is more complete.
So, if this could be develop I'm pretty sure companies around the world could get really mad at the project because it will be even better than all their services and will be free, that is a huge bomb for everyone. I have thought about this for years even I created a demo dashboard using Angular JS but then I moved out to another country and also Angular launched Angular 2+ and everything changed. Also firebase realtime database is something that I also had in mind.

**Comment by djbo1982** [comment]
@hackerunet You are rights, we need to think in all, because the fury of big companies can come to us...

I think we can develop for us, @netluca1 i can paid some money, because this is for me not for sale, for my partners, and you @netluca1 ? we need to have a mid point for this implementation, because we can shared to traccar project to @tananaev, and maybe the people hows want this need to compilate a custom version with the dashboard, the mobile parts is important to, because with new languages we can create ones for web and mobile version.

For the language, i don't have problems with angular (google language)i thinks is one of the best option to develop at this time.

We have a best project in backend a complete core with many functions... can you put a capture of your old client project?

By the way, with the new interfaces we can implement some things like turn by turn to find a car from client interface

**Comment by hackerunet** [comment]
I think the best could be give it to anton and release it. My old UI is in http://www.axiomagps.com

**Comment by djbo1982** [comment]
Is the best option!

Well! What we go to do? Let’s defining this...

**Comment by djbo1982** [comment]
Hello to all in this discussion... @hackerunet @netluca1 @oliv3...

I want to make this develop, and i need help, i can help with money

**Comment by netluca1** [comment]
Hello to all @hackerunet; @oliv3; @djbo1982; @cdmarioe; news for integration?
have you heard Anton too?
If not, I would suggest proposing this integration to him.
@tananaev what do you think?

@oliv you have finish integration in grafana? If yes is possible integrate in Traccar?

**Comment by tananaev** [comment]
Think about what?

**Comment by hackerunet** [comment]
Hi there, not enough time to work on a new dashboard by my side. I have
different projects to work on. But, the most ideal scenario would ve to
have the ui on angular or reactive. regards.

**Comment by djbo1982** [comment]
Ok, I say long time ago! If some body want to develop this write me!! I can paid for this dashboard and other things I need from this big an important project

**Comment by tananaev** [comment]
Send me an email to discuss it.

**Comment by netluca1** [comment]
@oliv3 what is your email?

**Comment by oliv3** [comment]
Check my profile.

**Comment by tananaev** [comment]
It's still an open idea.

**Comment by nabibou96** [comment]
The dashboard is a great idea, providing a brief summary of the fleet, especially for larger operations. In my case, I implemented a documentation check for each vehicle, allowing users to add expiration dates for insurance.

Imagine a well-organized table showing which vehicles need to renew their paperwork, along with their status, positions, and the assigned driver’s details, including their contact number. This could be a game-changer for fleet managers.

Similar examples can be applied to maintenance and other areas as well.
